internal/server/provider_commit_status_test.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2025 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"testing"
+
+	apiv1beta3 "github.com/fluxcd/notification-controller/api/v1beta3"
+	eventv1 "github.com/fluxcd/pkg/apis/event/v1beta1"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func Test_validateCommitStatusExpr(t *testing.T) {
+	validationTests := []string{
+		"(event.involvedObject.kind + '/' + event.involvedObject.name + '/' + event.metadata.foo + '/' + provider.metadata.uid.split('-')[0]).lowerAscii()",
+		"event.involvedObject.kind + '/' + provider.metadata.uid",
+		"event.metadata.foo + '/' + provider.metadata.uid",
+		"event.involvedObject.kind + '/' + event.metadata.foo + '/' + alert.metadata.name + '/' + provider.metadata.name",
+	}
+
+	for _, tt := range validationTests {
+		t.Run(tt, func(t *testing.T) {
+			g := NewWithT(t)
+			g.Expect(validateCommitStatusExpr(tt)).To(Succeed())
+		})
+	}
+}
+
+func Test_validateCommitStatusExpr_Invalid(t *testing.T) {
+	validationTests := []struct {
+		expression string
+		wantError  string
+	}{
+		{
+			"event.involvedObject.kind == 'Kustomization'",
+			"CEL expression output type mismatch: expected string, got bool",
+		},
+		{
+			"notification.metadata.name + '/' + provider.metadata.uid",
+			"undeclared reference to 'notification'",
+		},
+	}
+
+	for _, tt := range validationTests {
+		t.Run(tt.expression, func(t *testing.T) {
+			g := NewWithT(t)
+			g.Expect(validateCommitStatusExpr(tt.expression)).To(MatchError(ContainSubstring(tt.wantError)))
+		})
+	}
+}
```

Comment:
We usually test success and error cases in a single top-level Test func branching if the error is expected or not